lib/ProductOpener/Ingredients.pm

```diff
@@ -1960,6 +1960,76 @@ sub parse_ingredients_text_service ($product_ref, $updated_product_fields_ref) {
 						$between = "origins:" . $between;
 					}
 
+					# allergens or traces
+					# single allergen in parenthesis, for example: Krupica od durum pšenice (gluten) -> durum wheat semolina (gluten)
+					# more than a single allergen is handle just after
+					# in Japanese, 豚肉を含む (contains (を含む) pork (豚肉))
+					# 一部に卵・小麦・乳成分・大豆を含む (contains (を含む) parts of (一部に), eggs, wheat, milk, soybeans (卵・小麦・乳成分・大豆)
+					if (
+						(
+							($between !~ /$separators|$and/)
+							and (
+								exists_taxonomy_tag(
```

Comment:
This is problematic because the allergens taxonomy contains tons of synonyms that are in fact not synonyms of the allergen, but ingredients that contain that allergen (e.g. "en:milk" has synonyms for all kinds of cheeses).

That means that if we have "Cheese (parmigiano reggiano)", "parmigiano reggiano" will be in the allergens taxonomy. We do not want to remove it.

One possibility could be to canonicalize the ingredient in parenthesis using the ingredients taxonomy, and check whether it matches the exact allergen:  en:milk, or en:gluten.

Reply:
ah yeah, you got a point there. It will be an issue:
- for milk vs lactose, both can appear in parentheses for allergens :(
	https://world.openfoodfacts.org/product/0074175597009/blue-frosted-sugar-cookies-cleo-leo
	https://world.openfoodfacts.org/product/03003491216/classic-condensed-soup-cream-of-chicken

- Is it an option to... split allergens taxonomy in two? oO (1. the allergens and 2. the substances containing allergens) 
- Is it an option to make a list of reference allergens? (maybe directly in Ingredients.pm)
- Otherwise, I will have to roll back these changes and keep only condition for Japanese keyword

Reply:
I apply changes to do what you wrote. But again, that will not work...